### PR TITLE
Improve the devcontainer experience

### PR DESCRIPTION
**Pull request by felipecrs**
I decided to give a shot in the devcontainer today, and I realized some improvements could be made.

- Do not require users to manually build the website before building container
- Do not require users to manually build the base container (`make local`)
- And some other minor things

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -12,10 +12,11 @@ services:
     shm_size: "256mb"
     build:
       context: .
-      dockerfile: docker/Dockerfile.dev
+      dockerfile: docker/Dockerfile
+      target: dev
     devices:
       - /dev/bus/usb:/dev/bus/usb
-      - /dev/dri:/dev/dri # for intel hwaccel, needs to be updated for your hardware
+      # - /dev/dri:/dev/dri # for intel hwaccel, needs to be updated for your hardware
```
Comment (felipecrs) [inline docker-compose.yml:19]: As it does not work for all users, I suggest to be commented out by default. BTW it does not work in my WSL2 environment.

**Comment by NickM-27** [comment]
Personally, I would like to see the Dockerfiles stay separate. Much simpler to know which changes are for the main build vs just for the devcontainer. And then as other dockerfiles are being added like in #2548 the dev container will need to encapsulate all that as well getting more complicated.

**Comment by felipecrs** [comment]
I'm just not sure how to achieve the same result by splitting the Dockerfiles.

**Comment by NickM-27** [comment]
Yeah that makes sense, to avoid needing to make locally it'll have to be in the same file most likely

**Comment by felipecrs** [comment]
Unless by:

- https://github.com/docker/compose/issues/9461

Which is probably not going to be implemented any time soon.

File: docker/Dockerfile [hunk 2]
```diff
@@ -136,3 +134,56 @@ EXPOSE 1935
...
+RUN npm install
+
+COPY web/ ./
+RUN npm run build
+
+
+# Frigate Container
+FROM frigate-without-web
+
+COPY --from=build-web /work/dist web/
```
Comment (blakeblackshear) [inline docker/Dockerfile:165]: We also want the build output copied for the devcontainer. Often times, I am only working on backend features and don't want to run the dev server for the frontend.

Comment (felipecrs) [reply docker/Dockerfile:165]: I'm planning to solve that by setting creating proper VS Code tasks and debug configuration launch that would build the UI when pressing F5.

**Comment by felipecrs** [comment]
This is ready to review again, demo:


https://user-images.githubusercontent.com/29582865/201791172-a70eb852-9cf3-4f90-b577-c81a111ba377.mp4

File: docker/Dockerfile (hunk 2 above)
Comment (felipecrs) [reply docker/Dockerfile]: Addressed in https://github.com/blakeblackshear/frigate/pull/3492/commits/f4b2c3f37121e61ae113180e3a85827dc72f555c

**Comment by NickM-27** [comment]
I think this looks good to me upon trying it, running on macos all I need to do is comment out the devices in the docker-compose but this is already the case with todays container.

File: Makefile [hunk 3]
```diff
@@ -8,29 +8,26 @@ CURRENT_GID := $(shell id -g)
...
+	docker buildx build --tag frigate:latest --load .
 
-local:
-	DOCKER_BUILDKIT=1 docker build -t frigate -f docker/Dockerfile .
+build_web:
+	docker buildx build --target web-dist --output web/dist .
 
 amd64:
-	docker buildx build --platform linux/amd64 --tag blakeblackshear/frigate:$(VERSION)-$(COMMIT_HASH) --file docker/Dockerfile .
+	docker buildx build --platform linux/amd64 --tag blakeblackshear/frigate:$(VERSION)-$(COMMIT_HASH) .
```
[the remark this answers is not in the document]
Comment (felipecrs) [reply Makefile:18]: Through an environment variable, like IMAGE_REPO?

[the remark this answers is not in the document]
Comment (felipecrs) [reply Makefile]: Please check now, if that matched your expectations.

[the remark this answers is not in the document]
Comment (NateMeyer) [reply Makefile]: Yes, thank you. Is there reason to only have the `ghcr.io/` in the push tag, and not included in the variable?

File: Makefile [hunk 4]
```diff
@@ -2,35 +2,33 @@ default_target: local
 
 COMMIT_HASH := $(shell git log -1 --pretty=format:"%h"|tail -1)
 VERSION = 0.12.0
+IMAGE_OWNER ?= blakeblackshear
```
Comment (blakeblackshear) [inline Makefile:5]: Maybe instead of `IMAGE_OWNER` this should be `IMAGE_REPO` with the default set to `ghcr.io/blakeblackshear/frigate`

Comment (felipecrs) [reply Makefile]: Done.

**Comment by felipecrs** [comment]
The testing repo (https://github.com/blakeblackshear/frigate/pull/4368) is causing some issues now. At first, it comes with a nodejs newer than the one we wanted and broke `docker build`, but now it's interfering with the installation of the [vscode common utils](https://github.com/devcontainers/features/tree/main/src/common-utils). I'm still checking what I can do.

**Comment by blakeblackshear** [comment]
Did you see this suggestion? https://github.com/blakeblackshear/frigate/pull/4368#discussion_r1025866398

**Comment by felipecrs** [comment]
> Did you see this suggestion? https://github.com/blakeblackshear/frigate/pull/4368#discussion_r1025866398

I wasn't able to make such thing work like `apt-get install -t nodesource nodejs`. I don't think it can be used for such thing.

But I think I found a better solution, though. Check last commit.

File: Dockerfile [hunk 5]
```diff
@@ -65,7 +65,7 @@ RUN --mount=type=bind,from=wheels,source=/wheels,target=/wheels \
     gnupg \
     wget \
     procps \
-    unzip tzdata libxml2 xz-utils \
+    unzip locales tzdata libxml2 xz-utils \
```
Comment (felipecrs) [inline Dockerfile:68]: Just so you know, locales would get installed anyway as a dependency of the hwaccel packages. Installing before ensures it's installed from the regular repos and not from the testing repos.

**Comment by felipecrs** [comment]
Ok, should be ready to review again.

**Review by blakeblackshear — APPROVED** [review]

**Pull request merged**